[Feature]: update python to python3

### Requirement

I want to update the Python version used in Jaeger's build scripts.

so that
contributors can use the latest version of Python (Python 3) without encountering compatibility issues and errors during the build process. This will improve the overall development experience and ensure that the contribution process is smoother for all contributors.


### Problem

**Problem:**

Jaeger's current build scripts utilize the "python" command instead of "python3," causing obstacles when contributors with Python 3 try to execute the "make fmt" command. This leads to errors and confusion, disrupting the smooth contribution process and delaying code formatting.

**Impact:**

The use of "python" instead of "python3" in Jaeger's build scripts prevents contributors with Python 3 from running the "make fmt" command successfully. This discrepancy causes errors and hampers the timely execution of code formatting tasks, affecting the overall contribution workflow.

**for eg**

when i run `make fmt` before submitting PR according to contribution guidelines i get following output 

`
./scripts/import-order-cleanup.sh inplace
./scripts/import-order-cleanup.sh: line 5: python: command not found
make: *** [Makefile:154: fmt] Error 127`

to solve this i changed to older version of python used in **"./scripts/import-order-cleanup.sh"** to python3 

```
#!/bin/bash

set -e

python scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
    grep -v \
        -e thrift-gen \
        -e swagger-gen \
        -e thrift-0.9.2 \
        -e proto-gen \
        -e model.pb.go \
        -e model_test.pb.go \
        -e storage_test.pb.go
)
```
above is updated to following 

```
#!/bin/bash

set -e

python3 scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
    grep -v \
        -e thrift-gen \
        -e swagger-gen \
        -e thrift-0.9.2 \
        -e proto-gen \
        -e model.pb.go \
        -e model_test.pb.go \
        -e storage_test.pb.go
)

```

i suggest to do it for every script file which is using older version of python

for example 
```

scripts/import-order-cleanup.sh:
  4  
  5: python scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
  6      grep -v \

scripts/updateLicenses.sh:
  4  
  5: python scripts/updateLicense.py $(git ls-files "*\.go" | \
  6      grep -v \

```

### Proposal

i suggest to do it for every script file which is using older version of python

for example 
```

scripts/import-order-cleanup.sh:
  4  
  5: python scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
  6      grep -v \

scripts/updateLicenses.sh:
  4  
  5: python scripts/updateLicense.py $(git ls-files "*\.go" | \
  6      grep -v \

```

### Open questions

_No response_

--- a/Makefile
+++ b/Makefile
@@ -386,11 +386,11 @@ build-crossdock-fresh: build-crossdock-linux
 
 .PHONY: changelog
 changelog:
-	python3 ./scripts/release-notes.py --exclude-dependabot
+	./scripts/release-notes.py --exclude-dependabot
 
 .PHONY: draft-release
 draft-release:
-	python3 ./scripts/draft-release.py
+	./scripts/draft-release.py
 
 .PHONY: install-tools
 install-tools:

--- a/docker-compose/monitor/README.md
+++ b/docker-compose/monitor/README.md
@@ -90,7 +90,7 @@ this SPM Development Environment.
 
 Run the following commands to setup the Python virtual environment and install the Opentelemetry SDK:
 ```shell
-python -m venv .venv
+python3 -m venv .venv
 source .venv/bin/activate
 pip install -r requirements.txt
 ```

--- a/scripts/draft-release.py
+++ b/scripts/draft-release.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 import re
 import subprocess
 

--- a/scripts/import-order-cleanup.py
+++ b/scripts/import-order-cleanup.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 import argparse
 
 def cleanup_imports_and_return(imports):

--- a/scripts/import-order-cleanup.sh
+++ b/scripts/import-order-cleanup.sh
@@ -2,7 +2,7 @@
 
 set -e
 
-python scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
+./scripts/import-order-cleanup.py -o $1 -t $(git ls-files "*\.go" | \
     grep -v \
         -e thrift-gen \
         -e swagger-gen \

--- a/scripts/release-notes.py
+++ b/scripts/release-notes.py
@@ -7,7 +7,7 @@
 # Requires personal GitHub token with default permissions:
 #   https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
 #
-# Usage: python release-notes.py --help
+# Usage: ./release-notes.py --help
 #
 
 import argparse

--- a/scripts/release-notes.py
+++ b/scripts/release-notes.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # This script can read N latest commits from one of Jaeger repos
 # and output them in the release notes format:
 # * {title} ({author} in {pull_request})

--- a/scripts/updateLicense.py
+++ b/scripts/updateLicense.py
@@ -1,6 +1,4 @@
-from __future__ import (
-    absolute_import, print_function, division, unicode_literals
-)
+#!/usr/bin/env python3
 
 import logging
 import re

--- a/scripts/updateLicenses.sh
+++ b/scripts/updateLicenses.sh
@@ -2,7 +2,7 @@
 
 set -e
 
-python scripts/updateLicense.py $(git ls-files "*\.go" | \
+./scripts/updateLicense.py $(git ls-files "*\.go" | \
     grep -v \
         -e thrift-gen \
         -e swagger-gen \

--- a/scripts/utils/ids-to-base64.py
+++ b/scripts/utils/ids-to-base64.py
@@ -1,7 +1,10 @@
+#!/usr/bin/env python3
+
 import base64
 import os
 import re
 import sys
+
 
 def trace_id_base64(match):
     id = int(match.group(1), 16)
@@ -9,22 +12,23 @@ def trace_id_base64(match):
     b64 = base64.b64encode(hex.decode('hex'))
     return '"traceId": "%s"' % b64
 
+
 def span_id_base64(match):
     id = int(match.group(1), 16)
     hex = '%016x' % id
     b64 = base64.b64encode(hex.decode('hex'))
-    return '"spanId": "%s"' % b64
+    return f'"spanId": "{b64}"'
+
 
 for file in sys.argv[1:]:
-    print file
-    backup = '%s.bak' % file
+    print(file)
+    backup = f'{file}.bak'
     with open(file, 'r') as fin:
         with open(backup, 'w') as fout:
             for line in fin:
                 # line = line[:-1] # remove \n
                 line = re.sub(r'"traceId": "(.+)"', trace_id_base64, line)
                 line = re.sub(r'"spanId": "(.+)"', span_id_base64, line)
-                # print(line)
                 fout.write(line)
     os.remove(file)
     os.rename(backup, file)

